components/formats-api/src/loci/formats/FormatTools.java

```diff
@@ -405,6 +427,42 @@ public static int getIndex(String order, int zSize, int cSize, int tSize,
   }
 
   /**
+   * Gets the rasterized index corresponding to the given Z, C, T,
+   * ModuloZ, ModuloC, ModuloT coordinates.
+   *
+   * @param order Dimension order.
+   * @param zSize Total number of focal planes.
+   * @param cSize Total number of channels.
+   * @param tSize Total number of time points.
+   * @param moduloZSize Total number of ModuloZ planes.
+   * @param moduloCSize Total number of ModuloC planes.
+   * @param moduloTSize Total number of ModuloT planes.
+   * @param num Total number of image planes (zSize * cSize * tSize),
+   *   specified as a consistency check.
+   * @param z Z coordinate of ZCTmZmCmT coordinate sextuple to convert to 1D index.
+   * @param c C coordinate of ZCTmZmCmT coordinate sextuple to convert to 1D index.
+   * @param t T coordinate of ZCTmZmCmT coordinate sextuple to convert to 1D index.
+   * @param moduloZ ModuloZ coordinate of ZCTmZmCmT coordinate sextuple to convert to 1D index.
+   * @param moduloC ModuloC coordinate of ZCTmZmCmT coordinate sextuple to convert to 1D index.
+   * @param moduloT ModuloT coordinate of ZCTmZmCmT coordinate sextuple to convert to 1D index.
+   */
+  public static int getIndex(String order,
+    int zSize, int cSize, int tSize,
+    int moduloZSize, int moduloCSize, int moduloTSize,
+    int num,
+    int z, int c, int t,
+    int moduloZ, int moduloC, int moduloT) {
+    return getIndex(order,
+                    zSize,
+                    cSize,
+                    tSize,
+                    num,
+                    (z * moduloZSize) + moduloZ,
+                    (c * moduloCSize) + moduloC,
+                    (t * moduloTSize) + moduloT);
```

Comment:
Sorry, I should have been clearer.  The FormatTools helper functions (both modulo and non-modulo) take the full real sizes for both zSize/cSize/tSize and the modulo sizes.  It needs the full size for the "num" check.  It could be passed the effective size and then compute the real size from it, but I retained the original intent to keep it similar to the original (and it can pass these values directly to the original), and because that's the information the caller has in practice; otherwise we'd need to compute the effective size, pass that and then immediately recompute the real size again.

The effective sizes are only used for the z/c/t indexes which follow.

I've pushed a commit which addresses this by clearly documenting what's used for each parameter for all the original and added methods.  Hope that's OK.